### PR TITLE
(MODULES-10879) Implement configuration management

### PR DESCRIPTION
**Pull request by reidmv**
I'm proposing this feature as a result of not being able to find a clear, supportable, and documented way to advise a PE customer how to manage agent configuration.

This commit introduces a parameter that can be used to manage agent configuration. This parameter is restricted to managing only specifically permitted settings. Initially, only runinterval.

Without this functionality, users can only manage agent configuration by independently writing their own ini_setting resources in their own code. This is inefficient and error-prone compared to supporting basic configuration management of the agent in the official Puppet agent supported module.

By supporting basic configuration management of settings in this module it will be easier and safer to configure the Puppet agent, as well as becoming easier to document a correct and maintainable way of doing so.

### CODE REVIEW

**Comment by reidmv** [comment]
See also: [MODULES-9260](https://tickets.puppetlabs.com/browse/MODULES-9260)

**Comment by kenyon** [comment]
Just yesterday I was looking for a way to do this, specifically because I want to enable splay on the agents in my Puppet Enterprise environment.

File: spec/classes/puppet_agent_configure_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    let(:pre_condition) do
+      [ super(),
+        'class puppet_agent { $config = [ ] }',
+      ]
+    end
+
+    it { is_expected.to compile }
+  end
+
+  context 'with settings input' do
```
Comment (GabrielNagy) [inline spec/classes/puppet_agent_configure_spec.rb:24]: Should we have some tests that assert the settings' absence when `ensure => absent` (regardless of whether  `value` is set)?

**Review by GabrielNagy — COMMENTED** [review]
Thanks for the PR @reidmv! I had a comment about the tests, but from a code perspective this looks good to me.

The PR tests with nightly Puppet are expected to fail, but the tests with released Puppet gems seem to be valid: https://github.com/puppetlabs/puppetlabs-puppet_agent/pull/525/checks?check_run_id=2068620749. Maybe this is because of the dependency changes in init.pp.

Also, the commit check fails due to the following:

```
	This commit summary didn't match CONTRIBUTING.md guidelines:

		Update README high-level module description

	The commit summary (i.e. the first line of the commit message) should start with one of:
		(MODULES-<digits>) # this is most common and should be a ticket at tickets.puppet.com
		(docs)
		(docs)(DOCUMENT-<digits>)
		(packaging)
```

**Review by GabrielNagy — APPROVED** [review]
Thanks for updating the tests! I also ran the acceptance suite pipeline with this PR, and everything passes.

**Pull request merged**